src/networking.c

```diff
@@ -2363,7 +2406,7 @@ int processMultibulkBuffer(client *c) {
             }
 
             c->qb_pos = newline-c->querybuf+2;
-            if (!(c->flags & CLIENT_MASTER) && ll >= PROTO_MBULK_BIG_ARG) {
+            if (c->querybuf != shared_qb && !(c->flags & CLIENT_MASTER) && ll >= PROTO_MBULK_BIG_ARG) {
```

Comment:
Upon further review, this optimization can still be valid. For instance, if the client sends `$64000\r\n`, we can prepare a sufficiently large buffer for it. I will modify the code so that, in this case, the client will take ownership of the shared buffer.